commitizen/commands/commit.py

```diff
@@ -48,9 +51,23 @@ def read_backup_message(self) -> str | None:
     def prompt_commit_questions(self) -> str:
         # Prompt user for the commit message
         cz = self.cz
-        questions = cz.questions()
+        questions = [dict(question) for question in cz.questions()]
+
         for question in filter(lambda q: q["type"] == "list", questions):
             question["use_shortcuts"] = self.config.settings["use_shortcuts"]
+
+        # Import allowed modules for 'filter'
+        global commitizen
+        import commitizen.cz.utils
+
+        for question in filter(
+            lambda q: isinstance(q.get("filter", None), str), questions
+        ):
+            question_filter = [
+                multiple_line_breaker(question["filter"].replace("\\n", "\n"))
+            ]
+            question["filter"] = eval("\n".join(question_filter))
```

Comment:
Hmmm... I don't really like this kind of `eval` stuff. Would like to know how @woile and @noirbizarre  think

Reply:
I'm not the biggest fan of `eval` either, it's a better choice than `exec` of course, was looking for an alternative
but nothing quite so capable of what we need it to do here to replicate these behaviours.

Otherwise, statically coded things you can call with a single string was an idea,
but is way too specific and limited to cover the simple cases needed here.

---

Makes me thing I'd like to restore my original `.cz.yaml` implementation of custom Conventional Commits,
before / without my evolutions and changes, to share it here for reference.

**To give an idea of my currently used `filter` keys :** https://gitlab.com/RadianDevCore/tools/pre-commit-crocodile/-/blob/47068c6bce3758119db692ff74d0de9370ae199c/.cz.yaml#L133

Reply:
I think something like https://gitlab.com/RadianDevCore/tools/pre-commit-crocodile/-/blob/47068c6bce3758119db692ff74d0de9370ae199c/.cz.yaml#L147 is still a bit better than using `eval` (yep, we'll need to make it a python module), but would like to gather more ideas from others before we make any decision on this one

Reply:
That one is evaluated to the Python function too given it's all the same parser,
the initial alternative was `multiple_line_breaker`, for that one it's fine,
but it would need to somehow preexist for the other 2 for example, but if so and properly named, fine.

To sum up, no eval, but `multiple_line_breaker`, `required_validator_subject_strip` and `required_validator_scope` in `.utils`.
And strings of `if`, `elif`, `else Error` for bindings.